File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/managers/GUIManager.java
package com.dumbdogdiner.warrior.managers;

import com.dumbdogdiner.stickyapi.bukkit.gui.GUI;
import com.dumbdogdiner.warrior.Warrior;
import com.dumbdogdiner.warrior.api.managers.WarriorGUIManager;
import com.dumbdogdiner.warrior.gui.*;
import com.dumbdogdiner.warrior.gui.settings.LanguageGUI;
import com.dumbdogdiner.warrior.gui.settings.NotificationsGUI;
import com.dumbdogdiner.warrior.gui.settings.PrivacyGUI;
import com.dumbdogdiner.warrior.gui.settings.SettingsGUI;

import java.util.HashMap;

public class GUIManager implements WarriorGUIManager  {

    private HashMap<Class<?>, GUI> map = new HashMap<>();

    public void registerGUIs() {
        map.put(ArenaGUI.class, new ArenaGUI(5, "Select Arena", Warrior.getInstance()));
        map.put(KitGUI.class, new KitGUI(5, "Select a Kit", Warrior.getInstance()));
        map.put(DeathSoundGUI.class, new DeathSoundGUI());
        map.put(SettingsGUI.class, new SettingsGUI());
        map.put(LanguageGUI.class, new LanguageGUI());
        map.put(NotificationsGUI.class, new NotificationsGUI());
        map.put(PrivacyGUI.class, new PrivacyGUI());
        map.put(ParticleTrailGUI.class, new ParticleTrailGUI(Warrior.getInstance()));
    }

    public <T> T get(Class<T> guiClass) {
        return guiClass.cast(map.get(guiClass));
    }

}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/managers/GUIManager.kt
package com.dumbdogdiner.warrior.managers

import com.dumbdogdiner.warrior.api.managers.WarriorGUIManager
import java.util.HashMap
import com.dumbdogdiner.stickyapi.bukkit.gui.GUI
import com.dumbdogdiner.warrior.gui.ArenaGUI
import com.dumbdogdiner.warrior.gui.KitGUI
import com.dumbdogdiner.warrior.gui.DeathSoundGUI
import com.dumbdogdiner.warrior.gui.settings.SettingsGUI
import com.dumbdogdiner.warrior.gui.settings.LanguageGUI
import com.dumbdogdiner.warrior.gui.settings.NotificationsGUI
import com.dumbdogdiner.warrior.gui.settings.PrivacyGUI
import com.dumbdogdiner.warrior.gui.ParticleTrailGUI

class GUIManager : WarriorGUIManager {
    private val map = HashMap<Class<*>, GUI>()

    fun registerGUIs() {
        map[ArenaGUI::class.java] = ArenaGUI()
        map[KitGUI::class.java] = KitGUI()
        map[DeathSoundGUI::class.java] = DeathSoundGUI()
        map[SettingsGUI::class.java] = SettingsGUI()
        map[LanguageGUI::class.java] = LanguageGUI()
        map[NotificationsGUI::class.java] = NotificationsGUI()
        map[PrivacyGUI::class.java] = PrivacyGUI()
        map[ParticleTrailGUI::class.java] = ParticleTrailGUI()
    }

    override fun <T> get(guiClass: Class<T>): T {
        return guiClass.cast(map[guiClass])
    }

}